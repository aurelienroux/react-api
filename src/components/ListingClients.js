import React from "react";
import { Link } from "react-router-dom";

class ListingClients extends React.Component {
  state = {
    data: "",
    loading: true
  };

  async componentDidMount() {
    await fetch(
      "https://tracktik-challenge.staffr.com/clients?_page=1&_limit=10"
    )
      .then(response => response.json())
      .then(data => this.setState({ data, loading: false }));
    console.log("data: ", this.state.data);
  }

  render() {
    return (
      <div>
        {this.state.loading ? (
          <div>loading...</div>
        ) : (
          <div>
            {this.state.data.map(({ givenName, id }, index) => {
              return (
                <div key={index}>
                  <Link to={`site/${id}`}>
                    <div>{givenName}</div>
                    <div>{id}</div>
                    <hr></hr>
                  </Link>
                </div>
              );
            })}
          </div>
        )}
      </div>
    );
  }
}

export default ListingClients;
