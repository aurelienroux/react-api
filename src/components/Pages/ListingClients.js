import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import { Container, CircularProgress, Avatar } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import ArrowForwardIosIcon from "@material-ui/icons/ArrowForwardIos";

const useStyles = makeStyles({
  avatar: {
    margin: 10
  },
  bigAvatar: {
    margin: 10,
    width: 60,
    height: 60
  }
});

const ListingClients = () => {
  const classes = useStyles();
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetch(`https://tracktik-challenge.staffr.com/clients?_page=1&_limit=5`)
      .then(response => response.json())
      .then(data => setData(data))
      .then(() => setLoading(false));

    return () => console.log("unmounting...");
  }, []);

  return (
    <Container maxWidth="md" className="container listing-clients">
      <h1>Clients</h1>
      {loading ? (
        <div className="loader">
          <CircularProgress size={50} />
        </div>
      ) : (
        <div>
          {data.map(({ givenName, id, logo, tags }, index) => {
            return (
              <Link to={`client/${id}`} key={index}>
                <div className="item-card">
                  <div className="item-card__avatar">
                    <Avatar
                      alt={`${givenName} logo`}
                      src={logo}
                      className={classes.bigAvatar}
                    />
                  </div>
                  <div className="item-card__infos">
                    <div>{givenName}</div>
                    {tags.map((tag, index) => (
                      <span className="tag" key={index}>
                        {tag}
                      </span>
                    ))}
                  </div>
                  <div className="item-card__arrow">
                    <ArrowForwardIosIcon />
                  </div>
                </div>
              </Link>
            );
          })}
        </div>
      )}
    </Container>
  );
};

export default ListingClients;
