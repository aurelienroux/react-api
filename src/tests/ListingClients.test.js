import React from "react";
import { create } from "react-test-renderer";
import ListingClients from "../components//Pages/ListingClients";

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const listingClients = create(<ListingClients />);
    expect(listingClients.toJSON()).toMatchSnapshot();
  });
});
