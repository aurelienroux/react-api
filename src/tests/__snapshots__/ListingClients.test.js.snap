// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<div
  className="MuiContainer-root container listing-clients MuiContainer-maxWidthMd"
>
  <h1>
    Clients
  </h1>
  <div
    className="loader"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
